SignalServiceKit/src/Storage/KeychainStorage.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import SAMKeychain
+
+public enum KeychainStorageError: Error {
+    case failure(description: String)
+}
+
+// MARK: -
+
+@objc public protocol KeychainStorage: class {
+
+    @objc func string(forKey key: String, service: String) throws -> String
+
+    @objc func set(string: String, forKey key: String, service: String) throws
+
+    @objc func data(forKey key: String, service: String) throws -> Data
+
+    @objc func set(data: Data, forKey key: String, service: String) throws
+
+    @objc func remove(key: String, service: String) throws
+}
+
+// MARK: -
+
+@objc
+public class SSKKeychainStorage: NSObject, KeychainStorage {
+
+    @objc public static let sharedInstance = SSKKeychainStorage()
+
+    // Force usage as a singleton
+    override private init() {
+        super.init()
+
+        SwiftSingletons.register(self)
+    }
+
+    @objc public func string(forKey key: String, service: String) throws -> String {
```

Comment:
TIOLI, but I'd prefer to maintain the order of params that SAMKeychain uses, so `service` first then `key`

The service represents a "bag" of potentially multiple keys.

e.g. if you think about it as a dictionary access, you might write something like:

```
let s3Password = allPasswords["amazon"]["s3"]
```

Not:

```
let s3Password = allPasswords["s3"]["amazon"]
```

Reply:
Done. I think `TestAppContext` is now broken, but until more of these branches are merged I can't fix it.

Reply:
In general, I'd prefer that you make these PR's not predicated on the future presumption that future PR's are going to merged in a timely way. I'm hoping that each PR can be a useful atomic step forward. 

I know it takes more of *your* time to prepare PR's in that way, but I believe it allows for a better review process and cleaner git history.